mujoco/src/Base.hh

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_PHYSICS_MUJOCO_BASE_HH_
+#define GZ_PHYSICS_MUJOCO_BASE_HH_
+
+#include <mujoco/mjspec.h>
+#include <mujoco/mujoco.h>
+
+#include <cstddef>
+#include <gz/common/Console.hh>
+#include <gz/physics/Implements.hh>
+#include <iostream>
+#include <memory>
+
+namespace gz
+{
+namespace physics
+{
+namespace mujoco
+{
+
+// Forward declarations
+struct LinkInfo;
+struct ModelInfo;
+struct WorldInfo;
+
+struct ShapeInfo
+{
+  ShapeInfo(std::size_t _entityId, std::shared_ptr<LinkInfo> _linkInfo)
+      : entityId(_entityId), linkInfo(_linkInfo)
+  {
+  }
+  std::size_t entityId;
+  mjsGeom *geom;
+  std::string name;
+  std::weak_ptr<LinkInfo> linkInfo;
+};
+
+struct LinkInfo
+{
+  LinkInfo(std::size_t _entityId, std::shared_ptr<ModelInfo> _modelInfo)
+      : entityId(_entityId), modelInfo(_modelInfo)
+  {
+  }
+  std::size_t entityId;
+  mjsBody *body;
+  std::string name;
+  std::weak_ptr<ModelInfo> modelInfo;
```

Comment:
I see that you're using `weak_ptr` here, whereas bullet-featherstone [stores an `Identity`](https://github.com/gazebosim/gz-physics/blob/gz-physics9/bullet-featherstone/src/Base.hh#L198). I'm still reorienting myself to these data structures; do you see an advantage with `weak_ptr`?